File: UseCases/src/test/java/codes/seanhenry/mockgenerator/generator/templates/ThrowingTest.kt
package codes.seanhenry.mockgenerator.generator.templates

import codes.seanhenry.mockgenerator.entities.Method
import codes.seanhenry.mockgenerator.generator.MockTransformer

class ThrowingTest: MockGeneratorTestTemplate() {

  override fun build(generator: MockTransformer) {
    generator.add(
        Method.Builder("throwingMethod")
            .throws()
            .build(),
        Method.Builder("throwingReturnMethod")
            .returnType("String")
            .throws()
            .build(),
        Method.Builder("throwingClosure")
            .parameter("closure") { param ->
              param.type().function { it.throws() }
            }
            .build(),
        Method.Builder("throwingClosureArgument")
            .parameter("closure") { param ->
              param.type().function { func ->
                func.argument("String")
                    .returnType("(String)")
                    .throws()
              }
            }
            .build()
    )
  }
}

File: UseCases/src/test/resources/stub/ThrowingTest.swift
func throwingMethod() throws {
}
var stubbedThrowingReturnMethodResult: String! = ""
func throwingReturnMethod() throws -> String {
return stubbedThrowingReturnMethodResult
}
func throwingClosure(closure: () throws -> ()) {
}
func throwingClosureArgument(closure: (String) throws -> (String)) {
}